src/Symfony/Component/Process/Process.php

```diff
@@ -332,23 +333,25 @@ public function start(callable $callback = null, array $env = [])
             throw new RuntimeException(sprintf('The provided cwd "%s" does not exist.', $this->cwd));
         }
 
-        $this->process = @proc_open($commandline, $descriptors, $this->processPipes->pipes, $this->cwd, $envPairs, $options);
+        $this->process = @proc_open($commandline, $descriptors, $this->processPipes->pipes, $this->cwd, $envPairs, $this->options);
 
-        if (!\is_resource($this->process)) {
-            throw new RuntimeException('Unable to launch a new process.');
-        }
-        $this->status = self::STATUS_STARTED;
+        if (!$this->options['create_new_console']) {
```

Comment:
Adding this "if" means that in `create_new_console` mode, the process cannot be monitored and `getStatus()` returns dummy values.

The very side effect of this "if" is to disable the destructor. But this could be achieved more directly by wrapping the call to stop() in __desctruct() with `if ($this->options['create_new_console']) {`.

This leads another discussion: detaching a process from the Symfony Process object (aka allowing the process to continue to run while the $process object/the PHP engine stopped) is a separate feature to me. We should allow it on Linux also (not calling stop() in destruct when some flag is set.)

Reply:
Yes, I added that "if" in order to not wait for the process created with `create_new_console` to finish, evidently you can't get the status of a process that will run after the parent process has terminated. 

So it's not just that you should not call stop on that process but you should not check that process in any form, after you started it, as it may cause the main process to not exit if the child is still running. 

On Linux, this is already supported, because you don't need an option, is enough to use the `&` shell operator at the end of the command, and you can detach from the main process with no problem.

Reply:
How can I verify what you describe? As far as I tested correctly, I was able to monitor the status of the "sleep 2" command from the main program also.

> On Linux, this is already supported `&`

that's not really the same since then you cannot use an array to specify the command.

Reply:
OK I checked again now, it seems the check is necessary, `is_resource` will block the main process from exiting, here's a screen recording I made now [https://www.youtube.com/watch?v=8AvsYKefvrE](https://www.youtube.com/watch?v=8AvsYKefvrE), ran on PHP Version 7.4.4 for windows. 

On Linux I had this line `$process = new Process(['php', '../artisan', 'send:contact-email-notification', $db_msg->id, '&']);` from what I remember worked for me , I can check later on WSL if something similar like the line above works on Linux too.

Reply:
> it seems the check is necessary, is_resource will block the main process from exiting

I'd need to see the content of `test.php` to be sure (do you call `start()` or `run()`?). I very much doubt the currently proposed patch is correct (e.g. try using `run()` and you'll see bad failures if I tested correctly).

> On Linux I had this line

the last `&` will be escaped, which means the shell won't read it as go-to-the-background

Reply:
test.php contains
```
<?php
 
require_once __DIR__.'/vendor/autoload.php';

use Symfony\Component\Process\Process;
 
$process = new Process(['php', 'worker.php']);
$process->setOptions(["create_new_console" => true]);  
$process->disableOutput();
$process->start();

echo "Main Done";
```
worker.php contains:
```
<?php
sleep(3);
file_put_contents('test.txt',  date('Y/m/d H:i:s'));
echo("Worker done -> this output will be lost");
```
Normally you should never use run with `create_new_console`, because run is meant to wait for the process created, I guess you can disable the wait in the run function if `create_new_console` is set to avoid any unwanted error, but that means the user has not read the documentation of the component. 

I tested the above `test.php`, with `run` instead of `start` on Windows and as expected it will throw an exception because of the `wait` that's executed in the `run` function. 

I think the decision to disable or not the wait in the run method, is primarily a design question, from my perspective is fine either way. 

In the case of Linux, I tested earlier it seems that you don't need the &, because by default when using start() the main process will not wait for the child process, but there's a caveat, you still need to not check the process after you started it or will not run(maybe because the termination of the main process is to fast?)

For example this code on Linux:
```
<?php
 
require_once __DIR__.'/vendor/autoload.php';

use Symfony\Component\Process\Process;
$process = new Process(['php', 'worker.php']);
//$process->setOptions(["create_new_console" => true]);  
$process->disableOutput();
$process->start();

echo "Main Done";
```
Will run and terminate immediately with no errors but, worker.php won't get executed, if you uncomment the commented line, it will produce the desired effect meaning that the main process will finish immediately and the worker will be executed later. 

This raises a problem because, in both cases Windows or Linux the created process should not be checked after it was started but the `create_new_console` is only a windows option, and even if the above example works well on both OS-es the option must be named to something like 'background-process' or 'start-detached' to not confuse devs.

That is of course if you want to support such feature on Linux also.

Reply:
> Normally you should never use run with create_new_console

I'm not sure I agree with this statement. There might be other reasons to use `create_new_console` than detaching from the main process.

We need to figure out why the main process blocks in your case. Could it be because we keep the pipes open? Can you try closing them in the destructor in a "detached mode"? That logic would work for both windows and linux, and would need a new dedicated option (or method).

Reply:
Yeah, I tested it on Win and Linux it works, by changing the destructor, the only problem now, is one of design since `create_new_console` is an windows only flag, and there's probably a need of a new pseudo option documented, for creating a background process.

Reply:
> new pseudo option documented, for creating a background process.

`detach` => true/false` (would default to `$options['create_new_process'] ?? false`)?

Reply:
OK, I changed `setOptions()` ,please see if the code needs refactoring.